include/deal.II/base/std_cxx17/algorithm.h

```diff
@@ -17,6 +17,9 @@
 
 #include <deal.II/base/config.h>
 
+DEAL_II_WARNING(
+  "This file is deprecated. Simply use the corresponding C++14 header <algorithm>.")
```

Comment:
I think this ought to be C++17. Same for `optional.h` below, as well as the other files in this directory.

Reply:
Can if fix this in a follow-up if the CI passes?

Reply:
Yes.